packages/backend-modules/voting/lib/Submission.js

```diff
@@ -117,38 +136,60 @@ const createSubmissionsQuery = ({
       ],
     },
   }
+
   const mustValue = value && {
-    term: { 'resolved.answers.payload.value': value },
-  }
-  const mustSubmissionId = filters?.id && { term: { id: filters.id } }
-  const mustNotSubmissionId = filters?.not && { term: { id: filters.not } }
-  const mustSubmissionIds = filters?.submissionIds?.length && {
-    terms: { id: filters.submissionIds },
-  }
-  const mustNotSubmissionIds = filters?.notSubmissionIds?.length && {
-    terms: { id: filters.notSubmissionIds },
+    nested: {
+      path: 'resolved.answers',
+      query: { term: { 'resolved.answers.payload.value': value } },
+    },
   }
-  const mustAnsweredQuestionIds = filters?.answeredQuestionIds && {
+
+  const expectedAnswers = (filters?.answeredQuestionIds || [])
+    .map((questionId) => ({
+      questionId,
+    }))
+    .concat(filters?.answers || [])
+
+  const mustHaveAnswers = expectedAnswers && {
     bool: {
-      must: filters?.answeredQuestionIds.map((id) => ({
-        term: { 'resolved.answers.questionId': id },
-      })),
+      must: expectedAnswers.map(({ questionId, valueLength }) => {
```

Comment:
Could this be written in a more readable way?

Reply:
Probably, yes. Really depends on what you'd consider more readable.

I find Query DSL of ElasticSearch is what makes it rather hard to read.

```
must: expectedAnswers.map(({ questionId, valueLength }) => { questionId, valueLength }
```

Example

```
must: [
  { questionId: '123', valueLength: { min: 10 } }, 
  { questionId: '456' }, 
  { questionId: '789', valueLength: { max: 100 } }
]
```

But it then shovels this into Query DSL pattern of booleans and must nodes.

Or is destructuring throwing you off? Like this you'd find more readable?

```
must: expectedAnswers.map((expectedAnswer) => {
  const { questionId, valueLength } = expectedAnswer
  ...
}
```

Reply:
Would you find this more readable?

```js
  const mustHaveAnswers = expectedAnswers?.length && {
    bool: {
      must: expectedAnswers.map(createMustHaveNestedAnswer),
    },
  }
```

And then elswhere

```js
const createMustHaveNestedAnswer = ({ questionId, valueLength }) => ({
  nested: {
    path: 'resolved.answers',
    query: {
      bool: {
        must: [
          { term: { 'resolved.answers.questionId': questionId } },
          valueLength && {
            range: {
              'resolved.answers.resolved.value.length': {
                gte: Math.max(1, valueLength.min) || 1,
                lte: valueLength.max || undefined,
              },
            },
          },
        ].filter(Boolean),
      },
    },
  },
})
```

(Which is a bit like moving difficult part into another code section, making mapping part more readable, but not actual code.)

Reply:
Maybe I just need to understand Query DSL better... I think the destructured approach is good, would it be possible to separate where the `questionId` comes from, and where the `valueLength` comes from?

Reply:
Maybe something like this: 

```
  const mustHaveAnswers = expectedAnswers?.length && {
    bool: {
      must: expectedAnswers.map(({ questionId, valueLength }) => {
        return {
          nested: {
            path: 'resolved.answers',
            query: {
              bool: {
                must: [
                  questionIdTerm,
                  valueLengtRange
                ].filter(Boolean),
              },
            },
          },
        }
      }),
    },
  }


```